server/models/Document.ts

```diff
@@ -638,7 +630,15 @@ class Document extends ParanoidModel {
     const whereClause = `
   "searchVector" @@ to_tsquery('english', :query) AND
     "teamId" = :teamId AND
-    "collectionId" IN(:collectionIds) AND
+    ${
+      options.includeDrafts
+        ? collectionIds.length
+          ? '("collectionId" IN(:collectionIds) OR "collectionId" IS NULL) AND'
+          : '"collectionId" IS NULL AND'
+        : collectionIds.length
+        ? '"collectionId" IN(:collectionIds) AND'
+        : ""
```

Comment:
If I'm following correctly – If `includeDrafts: false` and you have access to no collections then the entire clause is removed here, giving you access to search everything in the team…

If that's true then we definitely need to add a test that would catch that.

Reply:
Grave mistake! Thanks for the spot :pray: I've simplified this a bit now and also added supporting tests.